##########################################################
# Parameter Configurations for TMCM-6214
##########################################################

/**:
  ros__parameters:
    ########## Interface Settings

    # Communication interface
    # Description: Interface used between the PC and the TMC
    # Possible values:
    # - 0: CAN
    comm_interface: 0

    # Communication interface name
    # Description: Name of the interface or device as detected by the PC
    # Possible values:
    # - can0, can1, can2, ..., etc
    comm_interface_name: can0

    # Communication interface Tx ID
    # Description: Currently applicable only for CAN interface (Tx ID)
    # Possible values:
    # - 0 to 255
    # Note: The ID that would work will be the one set at the TMC's GP: 71
    comm_tx_id: 1

    # Communication interface Rx ID
    # Description: Currently applicable only for CAN interface (Rx ID)
    # Possible values:
    # - 0 to 255
    # Note: The ID that would work will be the one set at the TMC's GP: 70
    comm_rx_id: 2

    # Communication timeout rate in seconds
    # Description: Indicates how long should the node will wait for the rx data
    # Possible values:
    # - value in millisecs 0 - 5000
    comm_timeout_ms: 100

    # Communication timeout tries
    # Description: Indicates how many the node will retry sending data before shutting down if no data received
    # Possible values:
    # - 1 - 3
    comm_exec_cmd_retries: 1

    ########## Additional Settings

    # Adhoc Mode
    # Description: This mode expects that the used module is not known. The velocity, position and torque relies on the additional ratios
    adhoc_mode: false

    # Active motors or axes
    # Description: Enables/disables active motors or axes. If disabled, settings for those motors will be ignored or not set.
    # Possible values:
    # - 0: Not used
    # - 1: Active
    en_motors: [1,1,1,1,1,1]

    # Auto start mode additional delay
    # Decription: Added delay if auto start mode is enabled. You can add delay if your TMCL program needs to run over 2 seconds.
    # Possible values: 0 - 10 (secs)
    auto_start_additional_delay: 0

##########################################################
##########################################################
# Configurations below are divided on each motor/axis. If the module used have more motors than listed below, 
# copy and paste all the configuration per motor, configure the needed topic names, wheel diameter values and
# generic ratios, and rename the motor name (should follow motor naming below - motor0, motor1, ..., motorn)
# otherwise, the node will fail.

    motor0:
      en_pub_tmc_info: true
      pub_rate_tmc_info: 10
      pub_actual_vel: true
      pub_actual_pos: true
      pub_actual_trq: true
      tmc_info_topic: "/tmc_info_0"
      tmc_cmd_vel_topic: "/cmd_vel_0"
      tmc_cmd_abspos_topic: "/cmd_abspos_0"
      tmc_cmd_relpos_topic: "/cmd_relpos_0"
      tmc_cmd_trq_topic: "/cmd_trq_0"

      wheel_diameter: 0.25
      additional_ratio_vel: 1.0
      additional_ratio_pos: 1.0
      additional_ratio_trq: 1.0

    motor1:
      en_pub_tmc_info: true
      pub_rate_tmc_info: 10
      pub_actual_vel: true
      pub_actual_pos: true
      pub_actual_trq: true
      tmc_info_topic: "/tmc_info_1"
      tmc_cmd_vel_topic: "/cmd_vel_1"
      tmc_cmd_abspos_topic: "/cmd_abspos_1"
      tmc_cmd_relpos_topic: "/cmd_relpos_1"
      tmc_cmd_trq_topic: "/cmd_trq_1"

      wheel_diameter: 0.25
      additional_ratio_vel: 1.0
      additional_ratio_pos: 1.0
      additional_ratio_trq: 1.0

    motor2:
      en_pub_tmc_info: true
      pub_rate_tmc_info: 10
      pub_actual_vel: true
      pub_actual_pos: true
      pub_actual_trq: true
      tmc_info_topic: "/tmc_info_2"
      tmc_cmd_vel_topic: "/cmd_vel_2"
      tmc_cmd_abspos_topic: "/cmd_abspos_2"
      tmc_cmd_relpos_topic: "/cmd_relpos_2"
      tmc_cmd_trq_topic: "/cmd_trq_2"

      wheel_diameter: 0.25
      additional_ratio_vel: 1.0
      additional_ratio_pos: 1.0
      additional_ratio_trq: 1.0

    motor3:
      en_pub_tmc_info: true
      pub_rate_tmc_info: 10
      pub_actual_vel: true
      pub_actual_pos: true
      pub_actual_trq: true
      tmc_info_topic: "/tmc_info_3"
      tmc_cmd_vel_topic: "/cmd_vel_3"
      tmc_cmd_abspos_topic: "/cmd_abspos_3"
      tmc_cmd_relpos_topic: "/cmd_relpos_3"
      tmc_cmd_trq_topic: "/cmd_trq_3"

      wheel_diameter: 0.25
      additional_ratio_vel: 1.0
      additional_ratio_pos: 1.0
      additional_ratio_trq: 1.0

    motor4:
      en_pub_tmc_info: true
      pub_rate_tmc_info: 10
      pub_actual_vel: true
      pub_actual_pos: true
      pub_actual_trq: true
      tmc_info_topic: "/tmc_info_4"
      tmc_cmd_vel_topic: "/cmd_vel_4"
      tmc_cmd_abspos_topic: "/cmd_abspos_4"
      tmc_cmd_relpos_topic: "/cmd_relpos_4"
      tmc_cmd_trq_topic: "/cmd_trq_4"

      wheel_diameter: 0.25
      additional_ratio_vel: 1.0
      additional_ratio_pos: 1.0
      additional_ratio_trq: 1.0

    motor5:
      en_pub_tmc_info: true
      pub_rate_tmc_info: 10
      pub_actual_vel: true
      pub_actual_pos: true
      pub_actual_trq: true
      tmc_info_topic: "/tmc_info_5"
      tmc_cmd_vel_topic: "/cmd_vel_5"
      tmc_cmd_abspos_topic: "/cmd_abspos_5"
      tmc_cmd_relpos_topic: "/cmd_relpos_5"
      tmc_cmd_trq_topic: "/cmd_trq_5"

      wheel_diameter: 0.25
      additional_ratio_vel: 1.0
      additional_ratio_pos: 1.0
      additional_ratio_trq: 1.0

